lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -41,6 +42,12 @@ def get_url(self, subsection):
     def get_show_grades(self, subsection):
         return subsection.show_grades(self.context['staff_access'])
 
+    def get_learner_has_access(self, subsection):
+        learner_is_full_access = self.context.get('learner_is_full_access', False)
+        course_blocks = self.context['course_blocks']
+        contains_gated_content = course_blocks.get_xblock_field(subsection.location, 'contains_gated_content')
+        return (not contains_gated_content) or learner_is_full_access
```

Comment:
I think mirroring this logic was a miss when I first built the completion summary (which counts locked units, complete units, incomplete units). Would it make sense to update this logic to match? https://github.com/edx/edx-platform/blob/11ec4a9aaa773c759fc0d2cb168a11ef08067590/lms/djangoapps/courseware/courses.py#L523-L527

Reply:
Upon examination, I don't think my logic is necessary. The `ContentTypeGateTransformer` transformer checks that for us, and just doesn't set any `contains_gated_content` values if the course isn't gated.

I've been cargo-culting this from some code that Cale wrote, but I think it was an unnecessary addition. Going to change this and see if any tests break...